Tales/Form.php

```diff
@@ -36,10 +36,13 @@ final class Form implements \PHPTAL_Tales
      */
     static public function calculateType($type)
     {
-        // This is done in two steps using the $nameParts intermediate variable
-        // because it causes a strict error if something other than a defined
-        // variable reference is passed to array_pop.
-        $nameParts = explode('_', $type);
+        if (strpos($type, '\\') !== false) {
+            $nameParts = explode('\\', $type);
+        } else {
+            // This is for legacy PSR-0 namespaced classes
+            $nameParts = explode('_', $type);
+        }
```

Comment:
Probably worth considering using `preg_split` here, so you can split on both characters at once. Then it'd also work for mixed names.